nix/runtime.nix

```diff
@@ -0,0 +1,64 @@
+{
+  # Plumbing tools:
+  closureInfo, runCommand, writeText, buildEnv,
+
+  # Actual dependencies to propagate:
+  bash, coreutils }:
+let
+  tools = buildEnv {
+    name = "lorri-runtime-tools";
+    paths = [ coreutils bash ];
+  };
+
+  runtimeClosureInfo = closureInfo {
+    rootPaths = [ tools ];
+  };
+
+  runtimeClosureInfoAsNix = runCommand "runtime-closure.nix" {}
+  ''
+    cat <<EOF > scratch.nix
+    # Nix with sandboxing requires every path used at build time be
+    # explicitly declared. If we simply passed in the paths, they
+    # would be copied in as sources. Using builtins.storePath we're
+    # able to tell Nix that, no, in fact, treat these not as sources
+    # to copy, but instead of a regular store path.
+    #
+    # Include the explicit closure, too, otherwise we'll get mysterious
+    # "file not found" errors due to the glibc interpreter being
+    # missing.
+    let
+      # Magic inspired by Nix's config.nix:
+      # https://github.com/NixOS/nix/blob/f9a2ea44867cd1dbb408bca4df0ced806137b7f7/corepkgs/config.nix.in#L23
+      #
+      # If the dependency is in the Nix store we're using, refer to
+      # it as a literal store path. If it isn't, refer to it "normally".
+      #
+      # This makes sandboxing happy when in a nix-build, and the
+      # evaluation happy when in a «cargo build».
+      tools_host = ${tools};
```

Comment:
Should annotate that this expands to the nix store path at creation of this nix file.